client/src/app/s-degree-plan-editor/s-degree-plan-editor.component.ts

```diff
@@ -0,0 +1,243 @@
+import { Component, OnInit } from "@angular/core";
+
+@Component({
+  selector: "app-s-degree-plan-editor",
+  templateUrl: "./s-degree-plan-editor.component.html",
+  styleUrls: ["./s-degree-plan-editor.component.css"],
+})
+export class SDegreePlanEditorComponent implements OnInit {
+  plan: any;
+  courseList: any;
+
+  /**
+   * The following changes to types were made to make it easier to display them on the frontend.
+   * They will be reversed when using them for backend calls.
+   *
+   * (semester) status: Type Changed from Number to String
+   * (semester) units: Added field to semesters (Type Number)
+   * (semester) difficulty: Type Changed from String to Number
+   * (semester) isAddingSemester: Boolean, flag to check if user wants to add a semester
+   */
+
+  constructor() {
+    /** Dummy data for plan used to demo markup */
+    this.plan = {
```

Comment:
Dummy data for now. I will use a service to map the data retrieved from the backend to match this format so that it's easier for the frontend to display them.